utils/storybook/StaticStatesTable.tsx

```diff
@@ -0,0 +1,151 @@
+import * as React from 'react';
+import styled from '@emotion/styled';
+import {StaticStates} from '@workday/canvas-kit-labs-react-core/lib/StaticStates';
+
+interface ComponentPropValue {
+  value: any;
+  label: string;
+}
+
+interface ComponentProps {
+  [key: string]: any;
+}
+
+/**
+ * A helper to generate a table of all possible states for component visual testing.
+ *
+ * This uses our `StaticStates` system. Note: This system requires the use of Canvas'
+ * `styled` wrapper within the component, instead of the default emotion one.
+ */
+export interface StaticStatesTableProps {
```

Comment:
I think more generically it would be a matrix of x props and y props. As it is now, "Props" represent a row and "state" (the className for the psuedo-selector state) are the columns.

More generically it is a matrix of prop combinations since `className` is really a prop.

I'm not sure I like the implementation as a permutation of props that has to be filtered. What about this instead?

```tsx
<ComponentPropsTable
  rowProps={[
    { label: 'Checked', checked: true, indeterminate: false },
    { label: 'Indeterminate', checked: false, indeterminate: true },
    // etc
  ]}
  columnProps={[
    { label: 'Focus', className: 'focus' },
    { label: 'Hover', className: 'hover' },
    // etc
  ]}
>
  (props) => <Checkbox {...props} />
</ComponentPropsTable>
```

You can always make some utility functions that will give you the `rowProps` array from permutations